src/main/java/com/couchbase/lite/util/Utils.java

```diff
@@ -165,78 +170,122 @@ public static String shortenString(String orig, int maxLength) {
+            }
         }
     }
 
     // from gzip
     public static int CHUNK_SIZE = 8192; // 1024 * 8
-    public static byte[] decompressByGzip(byte[] sourceBytes){
+
+    public static byte[] decompressByGzip(byte[] sourceBytes) {
         byte[] buffer = null;
         ByteArrayOutputStream out = null;
         ByteArrayInputStream in = null;
         GZIPInputStream gzip = null;
         try {
-            out    = new ByteArrayOutputStream();
+            out = new ByteArrayOutputStream();
             buffer = new byte[CHUNK_SIZE];
-            in     = new ByteArrayInputStream(sourceBytes);
-            gzip   = new GZIPInputStream(in);
+            in = new ByteArrayInputStream(sourceBytes);
+            gzip = new GZIPInputStream(in);
             int len = 0;
             while ((len = gzip.read(buffer, 0, CHUNK_SIZE)) != -1) {
                 out.write(buffer, 0, len);
             }
             return out.toByteArray();
-        }
-        catch (IOException ex){
+        } catch (IOException ex) {
             Log.w(Log.TAG, "Failed to decompress gzipped data: " + ex.getMessage());
             return null;
-        }
-        finally {
-            try{ if(out  != null) { out.close();  } } catch(IOException ex) {}
-            try{ if(in   != null) { in.close();   } } catch(IOException ex) {}
-            try{ if(gzip != null) { gzip.close(); } } catch(IOException ex) {}
+        } finally {
+            try {
+                if (out != null) {
+                    out.close();
+                }
+            } catch (IOException ex) {
+            }
+            try {
+                if (in != null) {
+                    in.close();
+                }
+            } catch (IOException ex) {
+            }
+            try {
+                if (gzip != null) {
+                    gzip.close();
+                }
+            } catch (IOException ex) {
+            }
         }
     }
 
-    public static Map<String, String> headersToMap(Header[] headers){
+    public static Map<String, String> headersToMap(Header[] headers) {
         Map<String, String> map = new HashMap<String, String>();
-        for(int i = 0; i < headers.length; i++){
+        for (int i = 0; i < headers.length; i++) {
             map.put(headers[i].getName(), headers[i].getValue());
         }
         return map;
     }
+
+    /**
+     * The following method shuts down an ExecutorService in two phases,
+     * first by calling shutdown to reject incoming tasks, and then calling shutdownNow,
+     * if necessary, to cancel any lingering tasks:
+     * http://docs.oracle.com/javase/6/docs/api/java/util/concurrent/ExecutorService.html
+     */
+    public static void shutdownAndAwaitTermination(ExecutorService pool) {
+        pool.shutdown(); // Disable new tasks from being submitted
+        try {
+            // Wait a while for existing tasks to terminate
+            if (!pool.awaitTermination(60, TimeUnit.SECONDS)) {
```

Comment:
In database.close() above, we are waiting for 20 seconds but here we are waiting for 60 seconds to force termination. Should the number here lower that 20 seconds or should the database.close() wait for me than 60 seconds?